### PR TITLE
Create basic CRUD for task and its respective specs

### LINKED ISSUE #21: Create options to go to tasks ( crud page )

### PR DESCRIPTION
**Pull request by diegokloop**
#### What does this PR do?

Create basic CRUD for task model

#### Where should the reviewer start?

`app/controllers/tasks_controller.rb`

#### How should this be manually tested?

Test the create, read, update and delete actions for task under`projects/1/tasks`

#### Any background context you want to provide?

...

#### What are the relevant issues?

Closes #21 

#### Screenshots (if appropriate)

...

#### Checklist:

* [x] If you've fixed a bug or added code that should be tested, add tests!

---

#### Database changes

...

#### Deployment instructions (Specific order, ENV variables, etc...)

...

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+# Timer controller logic
+class TasksController < ApplicationController
+  def initialize
```
Comment (fede-moya) [inline app/controllers/tasks_controller.rb]: ????? wtf

File: app/views/tasks/_back_to_index.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<span>
```
Comment (fede-moya) [inline app/views/tasks/_back_to_index.html.erb]: el index esta mismo en el panel no es necesario

Comment (lucaleivaloop) [reply app/views/tasks/_back_to_index.html.erb]: @fede-moya, as is, the button is necessary, there will be a task to combine the `tasks` views into the `project` tab, during that task this button could be erased, but for now, if someone needs to use any of the views this button makes the navigation much simpler

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      redirect_to project_tasks_path(@project.id)
+    else
+      flash[:error] = @task.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
+  def destroy
+    if @task.destroy
+      flash[:success] = 'Task delete correctly!'
```
[the remark this answers is not in the document]
Comment (lucaleivaloop) [reply app/controllers/tasks_controller.rb]: Good catch, ready to upload, just waiting to see if anything else comes up

File: app/controllers/tasks_controller.rb [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  end
+
+  private
+
+  def tasks_params
+    permited_params =
+      params
+      .require(:task)
+      .permit(:name, :state, :due_date, :start_date, :priority, :project_id)
+    permited_params[:project_id] = @project.id
```
Comment (RodrigoGarciaLab) [inline app/controllers/tasks_controller.rb]: :project_id appears twice. Is this somehow desired?

**Pull request merged**